modules/home-manager/neovim.nix

```
@@ -7,10 +7,38 @@
 
 let
   cfg = config.catppuccin.nvim;
+
+  toLuaValue =
+    value:
+    if builtins.isAttrs value then
+      toLuaTable value
+    else if lib.isString value || lib.isPath value then
+      ''"${toString value}"''
+    else if lib.isList value then
+      "{${lib.strings.concatStringsSep "," (map toLuaValue value)}}"
+    else if lib.isBool value then
+      if value == true then "true" else "false"
+    else
+      toString value;
+
+  toLuaTable =
+    attrset:
+    let
+      values = lib.strings.concatStringsSep "," (
+        lib.attrsets.mapAttrsToList (name: value: "${name} = ${toLuaValue value}") attrset
+      );
+    in
+    "{${values}}";
 in
 
 {
-  options.catppuccin.nvim = catppuccinLib.mkCatppuccinOption { name = "neovim"; };
+  options.catppuccin.nvim = catppuccinLib.mkCatppuccinOption { name = "neovim"; } // {
+    extraConfig = lib.mkOption {
+      description = "Extra settings that will be passed to the setup function.";
+      default = { };
+      type = lib.types.attrs;
```

if we do use the lua generator we can do something like this too
```suggestion
      type = lib.types.submodule { freeformType = lib.types.attrsOf lib.types.anything; };
```

---

Is there a difference between this and just using attrs?

---

Currently there would be no difference, *but* I would still encourage the use of freeform modules

From the [NixOS manual](https://nixos.org/manual/nixos/unstable/#sec-freeform-modules):

> Freeform modules allow you to define values for option paths that have not been declared explicitly. This can be used to add attribute-specific types to what would otherwise have to be attrsOf options in order to accept all attribute names.

Further, this would also allow us to specify common options (like a transparent background) cleanly in our documentation rather than relying completely on the port's docs